plugins/Tools/RotateTool/RotateTool.py

```diff
@@ -151,3 +156,99 @@ def setRotationSnapAngle(self, angle):
 
     def resetRotation(self):
         Selection.applyOperation(SetTransformOperation, None, Quaternion(), None)
+
+    def layFlat(self):
+        # Based on https://github.com/daid/Cura/blob/SteamEngine/Cura/util/printableObject.py#L207
+        # Note: Y & Z axis are swapped
+
+        self.operationStarted.emit(self)
+        progress_message = Message("Laying object flat on buildplate...", lifetime = 0, dismissable = False)
+        progress_message.setProgress(0)
+        progress_message.show()
+
+        selected_object = Selection.getSelectedObject(0)
```

Comment:
Any chance you can iterate through all selected objects? Now you completely ignore multi-selection.